regtests/bin/matrix_divider.sh

```diff
@@ -0,0 +1,78 @@
+#   submit the job using sbatch.                                              #
+#                                                                             #
+#                                                      Ali Abdolali           #
+#                                                      August 2018            #
+#                                                      March 2021             #
+#                                                                             #
+#    Copyright 2013 National Weather Service (NWS),                           #
+#       National Oceanic and Atmospheric Administration.  All rights          #
+#       reserved.  WAVEWATCH III is a trademark of the NWS.                   #
+#       No unauthorized use without permission.                               #
+#                                                                             #
+# --------------------------------------------------------------------------- #
+# --------------------------------------------------------------------------- #
+# 1.  clean up and definitions                                                #
+# --------------------------------------------------------------------------- #
+rm before
+rm list
+rm matrix?
+rm -r ../model?
+#Put the job requirement/spec in "before"
+sed -e "/run_test/,\$d" matrix > before
+#Put the list of tests in "list"
+command egrep 'run_test|ww3_tp2.14' matrix | cat >> list
+#Count the number of lines in "list"
+listn=$(wc -l < list)
+count=0
+countT=0
+# Define the number of test in each subset "maxlist"
+maxlist=100
+matrixno=1
+# --------------------------------------------------------------------------- #
+# 2.  Divide and dump in subsets                                              #
+# --------------------------------------------------------------------------- #
+while read line
+do
+if [[ count -eq 0 ]]
+then 
+cat before >> matrix$matrixno
+fi
+  if [[ count -le maxlist ]]
+     then
+     echo $line >> matrix$matrixno
+     (( count = count + 1 ))
+     (( countT = countT + 1 ))
+#     echo $count
+#     echo $countT
+ else
+ count=0
+ (( matrixno = matrixno + 1 ))
+  fi
+done < list
+
+  for i in `seq 1 1 "$((matrixno))"`; do
+#Replace matrix.out > matrix?.out, model > model?
+  cp -r ../model ../model$i
+  sed -i 's/'matrix.out'/'matrix${i}.out'/gI' matrix$i
+  sed -i 's/'model'/'model${i}'/gI' matrix$i
+  echo "  echo ' '"                                                                     >> matrix$i
+  echo "  echo '     **************************************************************'"   >> matrix$i
+  echo "  echo '     *  end of WAVEWATCH III matrix$i of regression tests         *'"   >> matrix$i
+  echo "  echo '     **************************************************************'"   >> matrix$i
+  echo "  echo ' '"                                                                     >> matrix$i
+sbatch matrix$i
```

Comment:
I would remove this as this would be machine dependent and this could then not be used by others who are not using slurm or if we port to another machine that does not use sbatch.

Reply:
Sure, I'll remove it.
Once other thing that I'll do for ourself is adding another script which does cloning the FB and Dev, download the tar from ftp, prepare matrix, divide it and submit. So I will include this to that.